### DISCUSSION TITLE
Setting block size on littlefs

### DISCUSSION THREAD

**Discussion by davidemania** [zephyrproject-rtos/zephyr#73967] [Q&A] [answered]
Hi,
it it possible to set the block size in Zephyr implementation of littlefs? I am experimenting with the filesystem mounted in a partition of main flash, and no matter what I try the block size is always 8192 bytes (0x2000). I would like to have a smaller one, to avoid wasting memory since most files to be stored are tiny. Is this for a reason, may be this size is imposed by hardware?

I read the docs but it looks like there is no config for this parameter, and in littlefs.h the macro FS_LITTLEFS_DECLARE_CUSTOM_CONFIG has no parameter besides the ones that can be found in Kconfig.

Browsing the sources it appears that a "const struct lfs_config" type is defined (in lfs.h) that holds a block_size field, and the readme.md shows an example where this field is used.

I can't figure out where this gets its value of 0x2000, and how to eventually change it, so any help would be very appreciated.

Processor is Atmel SAMV71Q21B, flash size 2MiB, RAM 384 KiB.

Best regards
Davide

**Comment by davidemania** [comment]
I just read the datasheet for the SAMV71 and the FLASH can be erased only in blocks of 8192 bytes, that could be a good reason for the above. It would be interesting to know where this value is stored, it is not in the devicetree as far as I know.

**Comment by ajarmouni-st** [comment, marked as answer]
Hi @davidemania, I don't think block size can be changed since it's a hardware characteristic of each Flash memory (@FRASTM can you confirm?). I believe it is stored in the Flash's parameter table. See https://www.macronix.com/Lists/ApplicationNote/Attachments/1870/AN114v1-SFDP%20Introduction.pdf

**Comment by ajarmouni-st** [comment]
LittleFs complete set of devicetree parameters is in `dts/bindings/fs/zephyr,fstab,littlefs.yaml` & `dts/bindings/fs/zephyr,fstab-common.yaml`.

**Reply by davidemania** [reply]
Thank you, I found the relevant parameter here:

atmel,sam-flash-controller
erase-block-size = < 0x2000 >;
